### DISCUSSION TITLE
Is possible to program a blue pill at least via serial using only Free(as freedom) Software tools?

### DISCUSSION THREAD

**Discussion by amreo** [stm32duino/Arduino_Core_STM32#2465] [Q&A] [answered]
Hi!
Is possible to program a STM32F103C6T8 board like blue pill via UART or serial using using only free/open source software tools?

**Comment by fpistm** [comment]
You can use the STM32CubeProgrammer to download over UART.

**Reply by amreo** [reply]
Isn't it proprietary software?

**Reply by fpistm** [reply]
![image](https://github.com/user-attachments/assets/9b61cfad-0ebf-49ab-825c-559054092a23)

**Reply by amreo** [reply]
Free download doesn't mean Free(as _freedom_) Software

**Reply by fpistm** [reply]
Right feel free to not use it but I have no other tools to advise. Probably it already exists such kind of tools but honestly don't see the need and this one is the official one.
You can implement one following the bootloader specification: https://www.st.com/resource/en/application_note/an3155-usart-protocol-used-in-the-stm32-bootloader-stmicroelectronics.pdf

You can try also to ask on the forum.

**Reply by fpistm** [reply, marked as answer]
Follow here: https://www.stm32duino.com/viewtopic.php?p=14711&hilit=free#p14711
